### PR TITLE
Add what a year blogpost

### PR DESCRIPTION
**Pull request by olblak**
Signed-off-by: Olblak <me@olblak.com>

A blogpost to go back on the year 2022

### CODE REVIEW

**Review by dduportal — APPROVED** [review]
:rocket: 👏

File: content/en/blog/2022/12/what_a_year.adoc [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+weight: 50
+images: [""]
+contributors: ["olblak"]
+---
+:hide-uri-scheme:
+
+== 2022 is over and what a year!
+
+2022 has been awesome for the Updatecli project.
+Updatecli went from a tool mainly used within the Jenkins project, to a project that others organizations could rely on to keep their git repositories up to date.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:14]:
```suggestion
Updatecli went from a tool mainly used within the Jenkins project to a project that others organizations could rely on to keep their git repositories up to date.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+Firstly, with link:https://news.ycombinator.com/item?id=30286047[HackerNews], then with link:https://youtu.be/157bsLD-0mM[CDcon], the project received many feedback and evolved. 2022, was rich in opportunities to understand how project maintainers benefited from Updatecli.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:16]:
```suggestion
Firstly, with link:https://news.ycombinator.com/item?id=30286047[HackerNews], then with link:https://youtu.be/157bsLD-0mM[CDcon], the project received many feedback and evolved a lot. 2022 was rich in opportunities to understand how project maintainers benefit from Updatecli.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+With the user base growing, came many feature requests and new update scenarios.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:18]:
```suggestion
With the user base growing came many feature requests and new update scenarios.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+Over the year, the Updatecli was:
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:20]:
```suggestion
Updatecli over this year:
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+- 45 releases
+- 15 new contributors
+- Many plugins addition, or improvement
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:24]:
```suggestion
- Many plugins additions and improvements
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+- A project Logo!
+- First (not expected) blog posts 
+
+image::/images/blog/2022/12/promaton.png["Promaton"]
+
+Several major ideas came to life
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:30]:
```suggestion
Several major ideas also came to life:
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+**Manifest validation**
+
+For decades, software projects maintained scripts written in many different languages to automate their project updates. All Updatecli's contributors maintained scripts like those. They also saw them being copy-paste across repositories up to a point where most of the logic was just technical debts that nobody knew about.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:34]:
```suggestion
For decades, software projects maintained scripts written in many different languages to automate their project updates. All Updatecli's contributors maintained scripts like those. They also saw them being copy-pasted across repositories up to a point where most of the logic was just technical debts that nobody knew about anymore.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+With Updatecli, the goal was to have a declarative way of describing those Updates, to reduce the complexity of maintaining those scripts.
+Unfortunately, with each new DSL comes to some learning curve.
+Using the Updatecli new DSL was error-prone, and maintaining the documentation, was time-consuming.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:38]:
```suggestion
Using this Updatecli new DSL was error-prone and maintaining the documentation was time-consuming.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+image::/images/blog/2022/12/promaton.png["Promaton"]
+
+Several major ideas came to life
+
+**Manifest validation**
+
+For decades, software projects maintained scripts written in many different languages to automate their project updates. All Updatecli's contributors maintained scripts like those. They also saw them being copy-paste across repositories up to a point where most of the logic was just technical debts that nobody knew about.
+
+With Updatecli, the goal was to have a declarative way of describing those Updates, to reduce the complexity of maintaining those scripts.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:36]:
```suggestion
With Updatecli, the initial goal was to get a declarative way of describing those updates to reduce the complexity of maintaining these scripts by creating a specific link:https://en.wikipedia.org/wiki/Domain-specific_language[DSL].
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+Unfortunately, with each new DSL comes to some learning curve.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:37]:
```suggestion
Unfortunately, with each new DSL comes some learning curve.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+Using the Updatecli new DSL was error-prone, and maintaining the documentation, was time-consuming.
+
+The project started generating and publishing "Jsonschema" on the "Jonschemastore" so everybody could benefit from validation and documentation, right from their IDE without any additional configurations.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:40]:
```suggestion
The project started generating and publishing a link:https://json-schema.org/[JSON Schema] on the link:https://www.schemastore.org/json/[JSON Schema Store] so everybody could benefit from validation and documentation right from their IDE without any additional configuration.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+All they had to do was to have their Updatecli manifest files live within `updatecli.d` directory. The Jsonschema also benefits the documentation website which means fewer things to maintain...
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:41]: > All they had to do was to have their Updatecli manifest files live within `updatecli.d` directory.

I don't understand the meaning and goal of this sentence, I would reword or omit it.

Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:41]:
```suggestion
All they had to do was to have their Updatecli manifest files live within `updatecli.d` directory. The Jsonschema also benefits the documentation website which means fewer things to maintain!
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+image::/images/blog/2022/12/pr1000.png["PR1000"]
+
+**GitHub action**
+
+Updatecli works great from a local machine, but it works even better when executed from CI environment.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:47]:
```suggestion
Updatecli works great from a local machine, but it works even better when executed from a CI environment.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+
+CI tooling offers many different kinds of trigger mechanisms that benefit Updatecli.
+Here some example
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:50]:
```suggestion
CI tooling offers many kinds of trigger mechanisms that benefit Updatecli.

Here are some examples:
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+
+* GitOps - triggered by a scheduler, Updatecli keeps GitOps repository up to date
+* Release - triggered by a release event, Updatecli can propagate a new release version quickly to downstream git repositories.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:53]:
```suggestion
* GitOps - triggered by a scheduler: Updatecli keeps GitOps repository up to date.
* Release - triggered by a release event: Updatecli can quickly propagate a new release version to downstream git repositories.
```

Comment (lemeurherve) [reply content/en/blog/2022/12/what_a_year.adoc:53]: > triggered by a scheduler:

To be explained? Or replaced by "a commit"?

File: content/en/blog/2022/12/what_a_year.adoc [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+
+Thanks, Joseph Petersen, who made the Updatecli GitHub Action, what it's today.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:55]:
```suggestion
Many thanks to link:https://github.com/jetersen[Joseph Petersen] who made link:https://github.com/updatecli/updatecli-action[the Updatecli GitHub Action] what it is today.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+
+image::/images/blog/2022/12/twitter.png["twitter"]
+
+**Autodiscovery**
+
+Updatecli started as a way to describe advanced Update scenarios using manifests. Of course for standard Update scenarios, manifests can be redundant, and sometimes a project needs a mix of both.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:61]:
```suggestion
Updatecli started as a way to describe advanced update scenarios using manifests. Of course for standard update scenarios, manifests can be redundant and sometimes a project needs a mix of both.
```
"both": both what?

Also, I like the old plural form of scenario but it's a matter of personal taste:
```suggestion
Updatecli started as a way to describe advanced update scenarii using manifests. Of course for standard update scenarios, manifests can be redundant and sometimes a project needs a mix of both.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+That's what Updatecli "Autodiscovery" is about.
+The purpose is to let Updatecli analyze files from a directory and then generate standard Update manifests before applying them. Autodiscovery is complementary to the manifest approach and is still an experiment.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:63]:
```suggestion
The purpose is to let Updatecli analyzes files from a directory and then generate standard update manifests before applying them. Autodiscovery is complementary to the manifest approach and is still experimental.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+
+Over the year, Updatecli started supporting automatic update for Dockerfile, Docker Compose file, Helm, Helmfile, Rancher Fleet, and Maven.
+
+Users can run one of the following commands from whatever directory to see what would change.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:67]:
```suggestion
Users can run one of the following commands from the directory of their choice to see what manifest(s) would be generated:
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+
+* `updatecli diff --experimental`
+* `updatecli manifest show --experimental`
+* `updatecli apply --experimental`
+
+
+**Updatemonitor**
+
+The Updatemonitor app was another well received-experiment.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:76]:
```suggestion
The Updatemonitor app is another well received experiment.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+image::/images/blog/2022/12/updatemonitor.png["Updatemonitor"]
+
+
+
+== Thank you all
+
+
+This year wouldn't have been possible without the many people who contributed to Updatecli.
+Thanks to Damien Duportal and Hervé Le Meur, for those endless discussions.
+Thanks, Joseph Petersen for being an early adopter and contributor of Updatecli.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:88]:
```suggestion
Thanks to link:https://github.com/dduportal[Damien Duportal] and link:https://github.com/lemeurherve[Hervé Le Meur] for those endless discussions.
Thanks lin:https://github.com/jetersen[Joseph Petersen] for being an early adopter and contributor of Updatecli.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 22]
```diff
@@ -0,0 +1,93 @@
...
+Thanks to everybody who has been part of the journey.
+
+2022 is over, See you in 2023.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:91]:
```suggestion
2022 is over, see you in 2023!
```

File: content/en/blog/2022/12/what_a_year.adoc (hunk 15 above)
Comment (olblak) [reply content/en/blog/2022/12/what_a_year.adoc:53]: I meant the cron scheduler and I thought that a scheduler is more generic

File: content/en/blog/2022/12/what_a_year.adoc [hunk 23]
```diff
@@ -0,0 +1,93 @@
...
+
+With the user base growing came many feature requests and new update scenarios.
+
+Updatecli over this year:
+
+- 45 releases
+- 15 new contributors
+- Many plugins additions and improvements
+- A project Logo!
+- First (not expected) blog posts
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:26]:
```suggestion
- First (unexpected) blog posts 
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+
+The Updatemonitor app is another well received experiment.
+While still in an early stage, the Updatemonitor app is a long-running service that leverages Updatecli plugins to monitor all kinds of information.
+
+image::/images/blog/2022/12/updatemonitor.png["Updatemonitor"]
+
+
+
+== Thank you all
+
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:85]:
```suggestion
== Thank you all
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+
+This year wouldn't have been possible without the many people who contributed to Updatecli.
+Thanks to link:https://github.com/dduportal[Damien Duportal] and link:https://github.com/lemeurherve[Hervé Le Meur] for those endless discussions.
+Thanks lin:https://github.com/jetersen[Joseph Petersen] for being an early adopter and contributor of Updatecli.
+Thanks to everybody who has been part of the journey.
+
+2022 is over, see you in 2023!
+
+Cheers
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:94]:
```suggestion
Cheers,
Olivier
```
or
```suggestion
Cheers,
link:https://github.com/Olblak[Olblak]
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 26]
```diff
@@ -0,0 +1,94 @@
...
+image::/images/blog/2022/12/updatemonitor.png["Updatemonitor"]
+
+
+
+== Thank you all
+
+
+This year wouldn't have been possible without the many people who contributed to Updatecli.
+Thanks to link:https://github.com/dduportal[Damien Duportal] and link:https://github.com/lemeurherve[Hervé Le Meur] for those endless discussions.
+Thanks lin:https://github.com/jetersen[Joseph Petersen] for being an early adopter and contributor of Updatecli.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:89]:
```suggestion
Thanks to link:https://github.com/jetersen[Joseph Petersen] for being an early adopter and contributor of Updatecli.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 27]
```diff
@@ -0,0 +1,94 @@
...
+* GitOps - triggered by a scheduler: Updatecli keeps GitOps repository up to date.
+* Release - triggered by a release event: Updatecli can quickly propagate a new release version to downstream git repositories.
+
+Many thanks to link:https://github.com/jetersen[Joseph Petersen] who made link:https://github.com/updatecli/updatecli-action[the Updatecli GitHub Action] what it is today.
+
+image::/images/blog/2022/12/twitter.png["twitter"]
+
+**Autodiscovery**
+
+Updatecli started as a way to describe advanced update scenarios using manifests. Of course for standard update scenarios, manifests can be redundant and sometimes a project needs a mix of both.
```
Comment (olblak) [inline content/en/blog/2022/12/what_a_year.adoc:62]:
```suggestion
Updatecli started as a way to describe advanced update scenarii using manifests. Of course for standard update scenario, a manifest can be redundant and sometimes a project needs a mix of both.
```

File: content/en/blog/2022/12/what_a_year.adoc [hunk 28]
```diff
@@ -0,0 +1,94 @@
...
+- A project Logo!
+- First (not expected) blog posts 
+
+image::/images/blog/2022/12/promaton.png["Promaton"]
+
+Several major ideas also came to life:
+
+**Manifest validation**
+
+For decades, software projects maintained scripts written in many different languages to automate their project updates. All Updatecli's contributors maintained scripts like those. They also saw them being copy-pasted across repositories up to a point where most of the logic was just technical debts that nobody knew about anymore.
```
Comment (lemeurherve) [inline content/en/blog/2022/12/what_a_year.adoc:34]:
```suggestion
For decades, software projects maintained scripts written in many different languages to automate their project updates. All Updatecli's contributors maintained scripts like those. They also saw them being copy-pasted across repositories up to a point where most of the logic was just technical debt that nobody knew about anymore.
```

File: content/en/blog/2022/12/what_a_year.adoc (hunk 25 above)
Comment (lemeurherve) [reply content/en/blog/2022/12/what_a_year.adoc:91]: I don't know if there is any better specific asciidoc syntax for contributor, in which case it should be used here and above if possible.

File: content/en/blog/2022/12/what_a_year.adoc [hunk 29]
```diff
@@ -0,0 +1,91 @@
...
+
+With Updatecli, the initial goal was to get a declarative way of describing those updates to reduce the complexity of maintaining these scripts by creating a specific link:https://en.wikipedia.org/wiki/Domain-specific_language[DSL].
+Unfortunately, with each new DSL comes some learning curve.
+Using this Updatecli new DSL was error-prone and maintaining the documentation was time-consuming.
+
+The project started generating and publishing a link:https://json-schema.org/[JSON Schema] on the link:https://www.schemastore.org/json/[JSON Schema Store] so everybody could benefit from validation and documentation right from their IDE without any additional configuration.
+All they had to do was to have their Updatecli manifest files live within `updatecli.d` directory. The Jsonschema also benefits the documentation website which means fewer things to maintain!
```
Comment (olblak) [inline content/en/blog/2022/12/what_a_year.adoc:41]:
```suggestion
Only requirement for IDE integration, is to have Updatecli manifest  located in the `updatecli.d` directory. The Jsonschema also benefits the documentation website which means fewer things to maintain!
```

Comment (olblak) [reply content/en/blog/2022/12/what_a_year.adoc:41]: @lemeurherve Is it better for you?

Comment (lemeurherve) [reply content/en/blog/2022/12/what_a_year.adoc:41]: Looks better indeed!

Additional suggestion:
```suggestion
The only requirement for the IDE integration is to have Updatecli manifest(s) located in the `updatecli.d` directory. The JSON Schema also benefits to the documentation website which means fewer things to maintain!
```

**Review by lemeurherve — APPROVED** [review]
🍾
